tldextract/tldextract.py

```diff
@@ -204,6 +218,18 @@ def __init__(self, cache_file=CACHE_FILE, suffix_list_urls=PUBLIC_SUFFIX_LIST_UR
         self.extra_suffixes = extra_suffixes
         self._extractor = None
 
+        self.cache_fetch_timeout = cache_fetch_timeout
+        if not isinstance(self.cache_fetch_timeout, (tuple, float,)):
+            try:
+                self.cache_fetch_timeout = float(self.cache_fetch_timeout)
+            except (ValueError, TypeError):
```

Comment:
I'm fine with failing fast here. If people bother to override it, they can bother to use the right type. I'll adjust.